src/App.tsx

```diff
@@ -1,61 +1,160 @@
+import React, { useState } from 'react';
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todoWithUsers: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState('');
+
+  const [languageError, setlanguageError] = useState(false);
+  const [isTitleError, setTitleError] = useState(false);
+  const [isUserError, setUserError] = useState(false);
+
+  const [visibleTodos, setVisibleTodos] = useState(todoWithUsers);
+
+  let maxId = Math.max(...todoWithUsers.map(todo => todo.id));
+
+  const createTodo = () => {
+    maxId += 1;
+
+    return ({
+      id: maxId,
+      title,
+      completed: false,
+      userId: +userId,
+      user: getUser(+userId),
+    });
+  };
+
+  function checkValue(inputValue: string) {
+    const reg = /[a-zA-Zа-яА-ЯГЄІЇ\d\s]/g;
+    const matches = inputValue.match(reg);
+
+    if (matches === null) {
+      setlanguageError(true);
+    }
+
+    return matches ? matches.join('') : '';
```

Comment:
Also will work. You can check it
```suggestion
    return matches?.join('') || '';
```

Reply:
thank you, has been fixed